### PR TITLE
chore(www): Refactor `homepage-newsletter`

### PR DESCRIPTION
**Pull request by fk**
Switch from template literal/string styles to [`sx`](https://theme-ui.com/sx-prop) and object styles.

### CODE REVIEW

**Comment by fk** [comment]
https://www.youtube.com/watch?v=QdRnRr7pLSk ✌️

File: www/src/components/homepage/homepage-newsletter.js [hunk 1]
```diff
@@ -11,100 +12,90 @@ import { mediaQueries } from "gatsby-design-tokens/dist/theme-gatsbyjs-org"
...
-        <Name>The Gatsby Newsletter</Name>
-        <Title>Keep up with the latest things Gatsby!</Title>
+        <h3 sx={nameStyles}>The Gatsby Newsletter</h3>
+        <h1 sx={titleStyles}>Keep up with the latest things Gatsby!</h1>
       </header>
-      <Form
+      <EmailCaptureForm
         isHomepage={true}
         confirmMessage="Success! You have been subscribed to the Gatsby newsletter. Expect to see a newsletter in your inbox each Wednesday (or the equivalent of US Wednesday in your time zone)!"
+        sx={{ mt: [5, null, null, null, 0] }}
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-newsletter.js:96]: We should check that this actually compiles correctly. `EmailCaptureForm` isn't a raw tag so I'm not sure if the classes will propagate.

File: www/src/components/homepage/homepage-newsletter.js [hunk 2]
```diff
@@ -11,100 +12,90 @@ import { mediaQueries } from "gatsby-design-tokens/dist/theme-gatsbyjs-org"
...
-const Ornament = styled(`span`)`
-  left: -${p => p.theme.space[1]};
-  position: absolute;
-  top: -${p => p.theme.space[2]};
-`
+const ornamentStyles = {
+  left: -1,
+  position: `absolute`,
+  top: -2,
+}
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-newsletter.js:62]: I'd make this a component and put the icon directly in it.

```suggestion
function Ornament() {
  return <span sx={{
  left: -1,
  position: `absolute`,
  top: -2,
}}>
  <NewsletterOrnament />
</span>
}
```

Alternately, I think the svg can be styled directly so you might just be able to do:

```jsx
<NewsletterOrnament sx={{
  left: -1,
  position: `absolute`,
  top: -2,
}} />

File: www/src/components/homepage/homepage-newsletter.js [hunk 3]
```diff
@@ -11,100 +12,90 @@ import { mediaQueries } from "gatsby-design-tokens/dist/theme-gatsbyjs-org"
...
-      ${p => p.theme.space[stripedBorderHeight]}
-  );
-  position: relative;
+const containerStyles = {
+  bg: `newsletter.background`,
+  border: t => `1px solid ${t.colors.newsletter.border}`,
+  borderRadius: 2,
+  flexDirection: `column`,
+  mb: 8,
+  p: t => `calc(${t.space[8]} * 1.2)`,
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-newsletter.js:21]: Isn't this just space[9]?

40 * 1.2 = 48

File: www/src/components/homepage/homepage-newsletter.js (hunk 2 above)
Comment (tesseralis) [reply www/src/components/homepage/homepage-newsletter.js:62]: also: do negatives work with the theme-ui sx properties?

Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js:62]: > also: do negatives work with the theme-ui sx properties?

Yep, they def. do work! Checked when refactoring https://github.com/gatsbyjs/gatsby/pull/25333/files#diff-b8226183ef427c89d128d6d47d8c31dc + co. Can't fully remember but I think that didn't work "before".

(… got curious:)

* https://github.com/system-ui/theme-ui/search?q=negative&unscoped_q=negative
* also mentioned in the docs now: https://theme-ui.com/sx-prop/#margin-and-padding  
  > Margin and padding are treated as first-class citizens in Theme UI, using values defined in the theme.space scale, and include an optional shorthand syntax for using negative space in your application.

File: www/src/components/homepage/homepage-newsletter.js (hunk 3 above)
Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js:21]: LOL yes :)

File: www/src/components/homepage/homepage-newsletter.js [hunk 4]
```diff
@@ -11,100 +12,90 @@ import { mediaQueries } from "gatsby-design-tokens/dist/theme-gatsbyjs-org"
...
+  pb: t =>
```
Comment (fk) [inline www/src/components/homepage/homepage-newsletter.js:22]: Also … out with `rhythm` :D

image

File: www/src/components/homepage/homepage-newsletter.js (hunk 3 above)
Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js]: 🙏

File: www/src/components/homepage/homepage-newsletter.js (hunk 2 above)
Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js]: > I think the svg can be styled directly

Went for that ✌️

File: www/src/components/homepage/homepage-newsletter.js (hunk 1 above)
Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js:93]: Threw in some color, LGTM?!

image

File: www/src/components/homepage/homepage-newsletter.js [hunk 5]
```diff
@@ -1,110 +1,98 @@
...
+const titleStyles = {
+  color: `newsletter.heading`,
+  fontSize: 4,
+  fontWeight: `heading`,
+  lineHeight: `dense`,
+  m: 0,
+  mt: 1,
+}
 
 const HomepageNewsletter = () => (
```
Comment (abhishekjakhar) [inline www/src/components/homepage/homepage-newsletter.js:76]: can we export it as a function declaration? we are doing it like this for all the other components which we are refactoring.

File: www/src/components/homepage/homepage-newsletter.js [hunk 6]
```diff
@@ -1,110 +1,98 @@
-import React from "react"
-import styled from "@emotion/styled"
+/** @jsx jsx */
+import { jsx } from "theme-ui"
+import { Flex } from "theme-ui"
```
Comment (abhishekjakhar) [inline www/src/components/homepage/homepage-newsletter.js:3]: can we club this with the jsx import above it?

File: www/src/components/homepage/homepage-newsletter.js (hunk 5 above)
Comment (abhishekjakhar) [inline www/src/components/homepage/homepage-newsletter.js:76]:
```suggestion
export default function homepageNewsletter() {}
```

File: www/src/components/homepage/homepage-newsletter.js (hunk 6 above)
Comment (fk) [inline www/src/components/homepage/homepage-newsletter.js:3]:
```suggestion
import { jsx, Flex } from "theme-ui"
```

Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js]: 👍

File: www/src/components/homepage/homepage-newsletter.js (hunk 5 above)
Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js]: 👍

File: www/src/components/homepage/homepage-newsletter.js [hunk 7]
```diff
@@ -1,111 +1,98 @@
...
+            position: `absolute`,
+            top: -2,
+          }}
+        />
+        <header>
+          <h3 sx={nameStyles}>The Gatsby Newsletter</h3>
+          <h1 sx={titleStyles}>Keep up with the latest things Gatsby!</h1>
+        </header>
+        <EmailCaptureForm
+          isHomepage={true}
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-newsletter.js:91]:
```suggestion
          isHomepage
```

File: www/src/components/homepage/homepage-newsletter.js [hunk 8]
```diff
@@ -1,111 +1,98 @@
...
-  margin-bottom: ${p => p.theme.space[8]};
-  padding: calc(${p => p.theme.space[8]} * 1.2);
-  padding-bottom: calc(
-    ${props => rhythm(props.theme.space[8] * 1.2)} +
-      ${p => p.theme.space[stripedBorderHeight]}
-  );
-  position: relative;
+const containerStyles = {
+  bg: `newsletter.background`,
+  border: t => `1px solid ${t.colors.newsletter.border}`,
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-newsletter.js:15]:
```suggestion
  border: 1,
  borderColor: `newsletter.border`
```

Comment (fk) [reply www/src/components/homepage/homepage-newsletter.js]: See 7196d72

**Comment by fk** [comment]
🙏 Ward!

**Pull request merged**